nsdb-cluster/src/multi-jvm/scala/io/radicalbit/nsdb/cluster/actor/MetadataSpec.scala

```diff
@@ -177,5 +181,142 @@ class MetadataSpec extends MultiNodeSpec(MetadataSpec) with STMultiNodeSpec with
+
+          metadataCoordinator ! GetMetricInfo("testDb", "testNamespace","animals")
+          expectMsg(MetricInfoGot("testDb", "testNamespace","animals", None))
+        }
+
+        awaitAssert {
+          metadataCoordinator ! GetMetricInfo("testDb", "testNamespace2","animals")
+          expectMsg(MetricInfoGot("testDb", "testNamespace2","animals", None))
+        }
+
+        awaitAssert {
+          metadataCoordinator ! GetMetricInfo("testDbWithInfo", "testNamespaceWithInfo","people")
+          expectMsg(MetricInfoGot("testDbWithInfo", "testNamespaceWithInfo","people", Some(MetricInfo("testDbWithInfo", "testNamespaceWithInfo","people",172800000,86400000))))
+        }
+      }
+
+      def checkSchemas(schemaCoordinator: ActorSelection) = {
+        awaitAssert{
+          schemaCoordinator ! GetSchema("testDb", "testNamespace", "people")
+          expectMsg(SchemaGot("testDb", "testNamespace", "people",
+            Some(
+              Schema("people",Map("city" -> SchemaField("city",DimensionFieldType,VARCHAR()), "timestamp" -> SchemaField("timestamp",TimestampFieldType,BIGINT()), "bigDecimalLong" -> SchemaField("bigDecimalLong",DimensionFieldType,BIGINT()), "bigDecimalDouble" -> SchemaField("bigDecimalDouble",DimensionFieldType,DECIMAL()), "value" -> SchemaField("value",ValueFieldType,DECIMAL()), "gender" -> SchemaField("gender",TagFieldType,VARCHAR())))
+            )
+          ))
+
+          schemaCoordinator ! GetSchema("testDb", "testNamespace", "animals")
+          expectMsg(SchemaGot("testDb", "testNamespace", "animals",
+            Some(
+              Schema("animals",Map("city" -> SchemaField("city",DimensionFieldType,VARCHAR()), "timestamp" -> SchemaField("timestamp",TimestampFieldType,BIGINT()), "bigDecimalLong" -> SchemaField("bigDecimalLong",DimensionFieldType,BIGINT()), "bigDecimalDouble" -> SchemaField("bigDecimalDouble",DimensionFieldType,DECIMAL()), "value" -> SchemaField("value",ValueFieldType,BIGINT()), "gender" -> SchemaField("gender",TagFieldType,VARCHAR())))
+            )
+          ))
+        }
+
+        awaitAssert{
+          schemaCoordinator ! GetSchema("testDb", "testNamespace2", "animals")
+          expectMsg(SchemaGot("testDb", "testNamespace2", "animals",
+            Some(
+              Schema("animals",Map("city" -> SchemaField("city",DimensionFieldType,VARCHAR()), "timestamp" -> SchemaField("timestamp",TimestampFieldType,BIGINT()), "bigDecimalLong" -> SchemaField("bigDecimalLong",DimensionFieldType,BIGINT()), "bigDecimalDouble" -> SchemaField("bigDecimalDouble",DimensionFieldType,DECIMAL()), "value" -> SchemaField("value",ValueFieldType,BIGINT()), "gender" -> SchemaField("gender",TagFieldType,VARCHAR())))
+            )
+          ))
+        }
+
+        awaitAssert{
+          schemaCoordinator ! GetSchema("testDbWithInfo", "testNamespaceWithInfo", "people")
+          expectMsg(SchemaGot("testDbWithInfo", "testNamespaceWithInfo", "people",
```

Comment:
this is a question actually...can you use expectMsg without the assert?

Reply:
yes it means that if no message is received or a message not equal to the specified one, it raises an exception, thus the test fails